From 49a013fd4066940d9713405988dbb8b3651cfe5c Mon Sep 17 00:00:00 2001
From: GammaC0de <gammac0de@users.noreply.github.com>
Date: Sun, 17 Oct 2021 01:43:19 +0300
Subject: [PATCH 02/73] remove deprecated BIGHTTPRequest

---
 src/pyload/plugins/downloaders/PornhubCom.py | 40 --------------------
 1 file changed, 40 deletions(-)

diff --git a/src/pyload/plugins/downloaders/PornhubCom.py b/src/pyload/plugins/downloaders/PornhubCom.py
index 65869179d..bb2b7659d 100644
--- a/src/pyload/plugins/downloaders/PornhubCom.py
+++ b/src/pyload/plugins/downloaders/PornhubCom.py
@@ -3,40 +3,11 @@
 import json
 import re
 
-from pyload.core.network.cookie_jar import CookieJar
-from pyload.core.network.exceptions import Abort
-from pyload.core.network.http.http_request import HTTPRequest
 from pyload.core.utils.misc import eval_js
 
 from ..base.simple_downloader import SimpleDownloader
 
 
-class BIGHTTPRequest(HTTPRequest):
-    """
-    Overcome HTTPRequest's load() size limit to allow loading very big web pages by
-    overrding HTTPRequest's write() function.
-    """
-
-    # TODO: Add 'limit' parameter to HTTPRequest in v0.6.x
-    def __init__(self, cookies=None, options=None, limit=1_000_000):
-        self.limit = limit
-        super().__init__(cookies=cookies, options=options)
-
-    def write(self, buf):
-        """
-        writes response.
-        """
-        if self.limit and self.rep.tell() > self.limit or self.abort:
-            rep = self.getResponse()
-            if self.abort:
-                raise Abort
-            with open("response.dump", mode="wb") as fp:
-                fp.write(rep)
-            raise Exception("Loaded Url exceeded limit")
-
-        self.rep.write(buf)
-
-
 class PornhubCom(SimpleDownloader):
     __name__ = "PornhubCom"
     __type__ = "downloader"
@@ -78,17 +49,6 @@ class PornhubCom(SimpleDownloader):
         self.resume_download = True
         self.multi_dl = True
 
-        try:
-            self.req.http.close()
-        except Exception:
-            pass
-
-        self.req.http = BIGHTTPRequest(
-            cookies=CookieJar(None),
-            options=self.pyload.request_factory.get_options(),
-            limit=2_000_000,
-        )
-
     def handle_free(self, pyfile):
         m = re.search(
             r'<div class="video-wrapper">.+?<script type="text/javascript">(.+?)</script>',
-- 
2.30.2

