--- a/src/pyload/core/config/default.cfg
+++ b/src/pyload/core/config/default.cfg
@@ -2,19 +2,19 @@ version: 2
 
 general - "General":
     en; language : "Language" = en
-    folder storage_folder : "Download folder" = ~/Downloads/pyLoad
-    bool debug_mode : "Debug mode" = True
+    folder storage_folder : "Download folder" = /opt/Downloads/pyLoad
+    bool debug_mode : "Debug mode" = False
     debug;trace;stack debug_level : "Debug level" = trace
     int min_free_space : "Minimum free space in MiB" = 1024
     bool folder_per_package : "Create folder for each package" = True
 permission - "Permissions":
     bool change_user : "Change user of running process" = False
-    str user : "Username for ownership" = user
+    str user : "Username for ownership" = root
     str folder : "Permission mode for created folders" = 0755
     bool change_file : "Change permissions of downloads" = False
     str file : "Permission mode for downloaded files" = 0644
     bool change_group : "Change group of running process" = False
-    str group : "Groupname for ownership" = users
+    str group : "Groupname for ownership" = root
     bool change_dl : "Change ownership of downloads" = False
 download - "Download":
     int chunks : "Maximum connections for one download" = 3
@@ -34,11 +34,11 @@ reconnect - "Reconnection":
 webui - "Web Interface":
     bool enabled : "Activated" = True
     bool use_ssl : "Use HTTPS" = False
-    bool develop : "Development mode" = True
+    bool develop : "Development mode" = False
     file ssl_certfile : "SSL Certificate" = ssl.crt
     file ssl_keyfile : "SSL Key" = ssl.key
     file ssl_certchain : "CA's intermediate certificate bundle (optional)" =
-    ip host : "IP address" = localhost
+    ip host : "IP address" = 0.0.0.0
     int port : "Port" = 8000
     Default;modern;pyplex theme : "Theme" = modern
     bool autologin : "Skip login if single user" = False
@@ -56,11 +56,11 @@ log - "Log":
     bool console_color : "Colorize console" = False
     bool syslog : "Sent log to syslog" = False
     local;remote syslog_location : "Syslog location" = local
-    folder syslog_folder : "Syslog local folder" =
+    folder syslog_folder : "Syslog local folder" = /opt/var/log
     ip syslog_host : "Syslog remote IP address" = localhost
     int syslog_port : "Syslog remote port" = 514
     bool filelog : "Save log to file" = True
     int filelog_size : "Maximum file size (in KiB)" = 5120
-    folder filelog_folder : "Log file folder" =
+    folder filelog_folder : "Log file folder" = /opt/var/log
     int filelog_entries : "Maximum log files" = 10
     bool filelog_rotate : "Log rotate" = True
