From e35c70d4c8d81b0bac0bde3ef35310a14cf6f8ec Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Mon, 20 Mar 2023 11:28:20 +0300
Subject: [PATCH] lang/luv: add as dependency

---
 lang/luv/Makefile | 27 +++++++++++++++------------
 1 file changed, 15 insertions(+), 12 deletions(-)

diff --git a/lang/luv/Makefile b/lang/luv/Makefile
index 3ee45d2d8..a4f2826c4 100644
--- a/lang/luv/Makefile
+++ b/lang/luv/Makefile
@@ -1,12 +1,14 @@
+# XXX bump
+
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=luv
-PKG_VERSION:=1.40.0-0
+PKG_VERSION:=1.43.0-0
 PKG_RELEASE:=$(AUTORELEASE)
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://github.com/luvit/luv/releases/download/$(PKG_VERSION)
-PKG_HASH:=24473a081c3928eec2a352369cbafda97059574f4a4276861274473e7c7d17a0
+PKG_HASH:=567a6f3dcdcf8a9b54ddc57ffef89d1e950d72832b85ee81c8c83a9d4e0e9de2
 
 PKG_MAINTAINER:=Morteza Milani <milani@pichak.co>
 PKG_LICENSE:=Apache-2.0
@@ -40,19 +42,20 @@ CMAKE_OPTIONS += \
 	-DWITH_LUA_ENGINE=$(if $(CONFIG_LUV_USE_LUAJIT_ENGINE),Luajit,Lua)
 
 define Build/InstallDev
-	$(INSTALL_DIR) $(1)/usr/include/luv
-	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/usr/include/luv/*.h $(1)/usr/include/luv
-	$(INSTALL_DIR) $(1)/usr/lib/lua
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libluv.so* $(1)/usr/lib
-	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/libluv.pc $(1)/usr/lib/pkgconfig
-	$(SED) 's,/usr/include,$$$${prefix}/include,g' $(1)/usr/lib/pkgconfig/libluv.pc
-	$(SED) 's,/usr/lib,$$$${exec_prefix}/lib,g' $(1)/usr/lib/pkgconfig/libluv.pc
+	$(INSTALL_DIR) $(1)/opt/include/luv
+	$(INSTALL_DATA) $(PKG_INSTALL_DIR)/opt/include/luv/*.h $(1)/opt/include/luv
+	$(INSTALL_DIR) $(1)/opt/lib/lua
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libluv.so* $(1)/opt/lib
+	$(INSTALL_DIR) $(1)/opt/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/pkgconfig/libluv.pc $(1)/opt/lib/pkgconfig
+	$(SED) 's,/opt/include,$$$${prefix}/include,g' $(1)/opt/lib/pkgconfig/libluv.pc
+	$(SED) 's,/opt/lib,$$$${exec_prefix}/lib,g' $(1)/opt/lib/pkgconfig/libluv.pc
 endef
 
 define Package/luv/install
-	$(INSTALL_DIR) $(1)/usr/lib/lua
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libluv.so.1.40.0 $(1)/usr/lib/lua/luv.so
+	$(INSTALL_DIR) $(1)/opt/lib/lua
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libluv.so.1.43.0 $(1)/opt/lib/lua/luv.so
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libluv.so.* $(1)/opt/lib
 endef
 
 
-- 
2.30.2

