From 8a0c3ce7df300fa655d0beaa2017be693ccd77d2 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Fri, 16 Dec 2022 17:26:45 +0300
Subject: [PATCH] libs/libxml2: fix args

---
 libs/libxml2/Makefile | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/libs/libxml2/Makefile b/libs/libxml2/Makefile
index bf99c3c71..44c186085 100644
--- a/libs/libxml2/Makefile
+++ b/libs/libxml2/Makefile
@@ -4,12 +4,13 @@
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
 #
+# XXX Entware specific: --with-catalog --with-legacy
 
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libxml2
 PKG_VERSION:=2.10.3
-PKG_RELEASE:=1a
+PKG_RELEASE:=1b
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=@GNOME/libxml2/$(basename $(PKG_VERSION))
@@ -74,13 +75,13 @@ CONFIGURE_ARGS += \
 	--enable-shared \
 	--enable-static \
 	--with-c14n \
-	--without-catalog \
+	--with-catalog \
 	--with-debug \
 	--with-html \
 	--without-ftp \
 	--without-http \
 	--without-iso8859x \
-	--without-legacy \
+	--with-legacy \
 	--with-output \
 	--without-pattern \
 	--without-push \
-- 
2.30.2

