From b478f81388ff2e674ef16f9956c0d074a4ff8ddb Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Sat, 10 Dec 2022 18:50:40 +0300
Subject: [PATCH] libs/libupnpp: add as dependency

---
 libs/libupnpp/Makefile | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/libs/libupnpp/Makefile b/libs/libupnpp/Makefile
index 5e1ce429c..c8839e624 100644
--- a/libs/libupnpp/Makefile
+++ b/libs/libupnpp/Makefile
@@ -38,17 +38,17 @@ and control points. It is shared by upmpdcli and upplay.
 endef
 
 define Build/InstallDev
-	$(INSTALL_DIR) $(1)/usr/include
-	$(CP) $(PKG_INSTALL_DIR)/usr/include/libupnpp $(1)/usr/include/
-	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libupnpp.{la,so*} $(1)/usr/lib/
-	$(INSTALL_DIR) $(1)/usr/lib/pkgconfig
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/pkgconfig/libupnpp.pc $(1)/usr/lib/pkgconfig/
+	$(INSTALL_DIR) $(1)/opt/include
+	$(CP) $(PKG_INSTALL_DIR)/opt/include/libupnpp $(1)/opt/include/
+	$(INSTALL_DIR) $(1)/opt/lib
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libupnpp.{la,so*} $(1)/opt/lib/
+	$(INSTALL_DIR) $(1)/opt/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/pkgconfig/libupnpp.pc $(1)/opt/lib/pkgconfig/
 endef
 
 define Package/libupnpp/install
-	$(INSTALL_DIR) $(1)/usr/lib
-	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libupnpp.so.* $(1)/usr/lib/
+	$(INSTALL_DIR) $(1)/opt/lib
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libupnpp.so.* $(1)/opt/lib/
 endef
 
 $(eval $(call BuildPackage,libupnpp))
-- 
2.30.2

