From ec4da6d001fb887dc080e663defe2ea45873f3b8 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Tue, 13 Dec 2022 16:17:25 +0300
Subject: [PATCH] net/samba4: add as dependency

---
 net/samba4/Makefile | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/net/samba4/Makefile b/net/samba4/Makefile
index 9c31ebf28..b8cea618d 100644
--- a/net/samba4/Makefile
+++ b/net/samba4/Makefile
@@ -454,6 +454,18 @@ define Package/samba4-server/conffiles
 /opt/etc/krb5.conf
 endef
 
+define Build/InstallDev
+	$(INSTALL_DIR) $(1)/opt/usr/include/samba-4.0
+	$(CP) $(PKG_INSTALL_DIR)/opt/include/samba-4.0/*.h $(1)/opt/usr/include/samba-4.0/
+
+	$(INSTALL_DIR) $(1)/opt/usr/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/pkgconfig/smbclient.pc $(1)/opt/usr/lib/pkgconfig/
+
+	$(SED) 's,^includedir=.*/include,includedir=$(STAGING_DIR)/opt/usr/include,; \
+		s,^libdir=.*/lib,libdir=$(STAGING_DIR_ROOT)/opt/lib,' \
+			$(1)/opt/usr/lib/pkgconfig/smbclient.pc
+endef
+
 $(eval $(call HostBuild))
 $(eval $(call BuildPackage,samba4-libs))
 $(eval $(call BuildPackage,samba4-server))
-- 
2.30.2

