From 3457c6734e73906639eae6e530c38617c650ae00 Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Fri, 9 Feb 2024 23:55:20 +0300
Subject: [PATCH] package/libs/libxml2: fix args

---
 package/libs/libxml2/Makefile | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/package/libs/libxml2/Makefile b/package/libs/libxml2/Makefile
index cce18f7b..ba55beeb 100644
--- a/package/libs/libxml2/Makefile
+++ b/package/libs/libxml2/Makefile
@@ -1,8 +1,10 @@
+# XXX Entware specific: -DLIBXML2_WITH_CATALOG=ON -DLIBXML2_WITH_LEGACY=ON
+
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libxml2
 PKG_VERSION:=2.12.4
-PKG_RELEASE:=1
+PKG_RELEASE:=1a
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.xz
 PKG_SOURCE_URL:=@GNOME/libxml2/$(basename $(PKG_VERSION))
@@ -98,7 +100,7 @@ CMAKE_HOST_OPTIONS += \
 CMAKE_OPTIONS += \
 	-DBUILD_SHARED_LIBS=ON \
 	-DLIBXML2_WITH_C14N=ON \
-	-DLIBXML2_WITH_CATALOG=OFF \
+	-DLIBXML2_WITH_CATALOG=ON \
 	-DLIBXML2_WITH_DEBUG=ON \
 	-DLIBXML2_WITH_FTP=OFF \
 	-DLIBXML2_WITH_HTML=ON \
@@ -106,7 +108,7 @@ CMAKE_OPTIONS += \
 	-DLIBXML2_WITH_ICONV=ON \
 	-DLIBXML2_WITH_ICU=OFF \
 	-DLIBXML2_WITH_ISO8859X=OFF \
-	-DLIBXML2_WITH_LEGACY=OFF \
+	-DLIBXML2_WITH_LEGACY=ON \
 	-DLIBXML2_WITH_LZMA=OFF \
 	-DLIBXML2_WITH_MEM_DEBUG=OFF \
 	-DLIBXML2_WITH_MODULES=OFF \
-- 
2.30.2

