From e7e672f7cd176d028054ec3aa63b505c232cc78a Mon Sep 17 00:00:00 2001
From: The-BB <tun.chen.bo@gmail.com>
Date: Fri, 16 Feb 2024 22:22:37 +0300
Subject: [PATCH] package/utils/e2fsprogs: add libe2p

---
 package/utils/e2fsprogs/Makefile | 33 ++++++++++++++++++++++++++++----
 1 file changed, 29 insertions(+), 4 deletions(-)

diff --git a/package/utils/e2fsprogs/Makefile b/package/utils/e2fsprogs/Makefile
index 913aff23..17510fe5 100644
--- a/package/utils/e2fsprogs/Makefile
+++ b/package/utils/e2fsprogs/Makefile
@@ -4,7 +4,7 @@
 # This is free software, licensed under the GNU General Public License v2.
 # See /LICENSE for more information.
 #
-# Entware specific: keep CONFIG_USE_GLIBC
+# Entware specific: keep CONFIG_USE_GLIBC; add libe2p (ext4magic)
 
 include $(TOPDIR)/rules.mk
 
@@ -35,7 +35,7 @@ define Package/e2fsprogs
   SUBMENU:=Filesystem
   TITLE:=Ext2/3/4 filesystem utilities
   URL:=http://e2fsprogs.sourceforge.net/
-  DEPENDS:=+libuuid +libext2fs
+  DEPENDS:=+libuuid +libext2fs +libe2p
 endef
 
 define Package/e2fsprogs/description
@@ -84,6 +84,19 @@ define Package/libcomerr/description
   bundled with e2fsprogs.
 endef
 
+# XXX for ext4magic
+define Package/libe2p
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE:=used by dumpe2fs, chattr, and lsattr
+  URL:=http://e2fsprogs.sourceforge.net/
+  DEPENDS:=+libuuid
+endef
+
+define Package/libe2p/description
+  This package contains libe2p, the library of the e2fsprogs package.
+endef
+
 define Package/tune2fs
 $(call Package/e2fsprogs)
   TITLE:=Ext2 Filesystem tune utility
@@ -207,6 +220,13 @@ define Build/InstallDev
 		$(PKG_BUILD_DIR)/lib/ss/ss.h \
 		$(PKG_BUILD_DIR)/lib/ss/ss_err.h \
 		$(1)/opt/include/ss/
+	# XXX for ext4magic
+	$(INSTALL_DIR) $(1)/opt/include/e2p
+	$(CP) $(PKG_INSTALL_DIR)/opt/include/e2p/e2p.h $(1)/opt/include/e2p
+	$(INSTALL_DIR) $(1)/opt/lib
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libe2p.{so,a}* $(1)/opt/lib
+	$(INSTALL_DIR) $(1)/opt/lib/pkgconfig
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/pkgconfig/e2p.pc $(1)/opt/lib/pkgconfig
 endef
 
 define Host/Compile
@@ -240,8 +260,6 @@ define Package/e2fsprogs/install
 	$(LN) e2fsck $(1)/opt/sbin/fsck.ext2
 	$(LN) e2fsck $(1)/opt/sbin/fsck.ext3
 	$(LN) e2fsck $(1)/opt/sbin/fsck.ext4
-	$(INSTALL_DIR) $(1)/opt/lib
-	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libe2p.so.* $(1)/opt/lib/
 	$(INSTALL_DIR) $(1)/opt/etc
 	$(INSTALL_DATA) ./files/e2fsck.conf $(1)/opt/etc/e2fsck.conf
 endef
@@ -251,6 +269,12 @@ define Package/libcomerr/install
 	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libcom_err.so.* $(1)/opt/lib/
 endef
 
+# XXX for ext4magic
+define Package/libe2p/install
+	$(INSTALL_DIR) $(1)/opt/lib
+	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libe2p.so.* $(1)/opt/lib/
+endef
+
 define Package/libss/install
 	$(INSTALL_DIR) $(1)/opt/lib
 	$(CP) $(PKG_INSTALL_DIR)/opt/lib/libss.so.* $(1)/opt/lib/
@@ -318,6 +342,7 @@ define Package/lsattr/install
 endef
 
 $(eval $(call BuildPackage,libcomerr))
+$(eval $(call BuildPackage,libe2p))
 $(eval $(call BuildPackage,libss))
 $(eval $(call BuildPackage,libext2fs))
 $(eval $(call BuildPackage,e2fsprogs))
-- 
2.30.2

