--- a/src/github.com/alist-org/alist/cmd/flags/config.go
+++ b/src/github.com/alist-org/alist/cmd/flags/config.go
@@ -2,6 +2,10 @@ package flags
 
 var (
 	DataDir     string
+	TempDir     string
+	IndexDir    string
+	LogPath     string
+	DBPath      string
 	Debug       bool
 	NoPrefix    bool
 	Dev         bool
--- a/src/github.com/alist-org/alist/cmd/root.go
+++ b/src/github.com/alist-org/alist/cmd/root.go
@@ -24,7 +24,7 @@ func Execute() {
 }
 
 func init() {
-	RootCmd.PersistentFlags().StringVar(&flags.DataDir, "data", "data", "config file")
+	RootCmd.PersistentFlags().StringVar(&flags.DataDir, "data", "/opt/var/lib/alist/data", "config file")
 	RootCmd.PersistentFlags().BoolVar(&flags.Debug, "debug", false, "start with debug mode")
 	RootCmd.PersistentFlags().BoolVar(&flags.NoPrefix, "no-prefix", false, "disable env prefix")
 	RootCmd.PersistentFlags().BoolVar(&flags.Dev, "dev", false, "start with dev mode")
--- a/src/github.com/alist-org/alist/internal/conf/config.go
+++ b/src/github.com/alist-org/alist/internal/conf/config.go
@@ -52,10 +52,10 @@ type Config struct {
 }
 
 func DefaultConfig() *Config {
-	tempDir := filepath.Join(flags.DataDir, "temp")
-	indexDir := filepath.Join(flags.DataDir, "bleve")
-	logPath := filepath.Join(flags.DataDir, "log/log.log")
-	dbPath := filepath.Join(flags.DataDir, "data.db")
+	tempDir := filepath.Join(flags.TempDir, "/opt/var/lib/alist/data/temp")
+	indexDir := filepath.Join(flags.IndexDir, "/opt/var/lib/alist/data/bleve")
+	logPath := filepath.Join(flags.LogPath, "/opt/var/log/alist.log")
+	dbPath := filepath.Join(flags.DBPath, "/opt/var/lib/alist/data/data.db")
 	return &Config{
 		Address:        "0.0.0.0",
 		Port:           5244,
