--- a/src/github.com/alist-org/alist/bootstrap/log.go
+++ b/src/github.com/alist-org/alist/bootstrap/log.go
@@ -27,7 +27,7 @@ func InitLog() {
 }
 
 func init() {
-	flag.StringVar(&conf.ConfigFile, "conf", "data/config.json", "config file")
+	flag.StringVar(&conf.ConfigFile, "conf", "/opt/var/lib/alist/data/config.json", "config file")
 	flag.BoolVar(&conf.Debug, "debug", false, "start with debug mode")
 	flag.BoolVar(&conf.Version, "version", false, "print version info")
 	flag.BoolVar(&conf.Password, "password", false, "print current password")
--- a/src/github.com/alist-org/alist/conf/config.go
+++ b/src/github.com/alist-org/alist/conf/config.go
@@ -43,12 +43,12 @@ func DefaultConfig() *Config {
 		Assets:      "/",
 		SubFolder:   "",
 		LocalAssets: "",
-		TempDir:     "data/temp",
+		TempDir:     "/opt/var/lib/alist/data/temp",
 		Database: Database{
 			Type:        "sqlite3",
 			Port:        0,
 			TablePrefix: "x_",
-			DBFile:      "data/data.db",
+			DBFile:      "/opt/var/lib/alist/data/data.db",
 		},
 		Cache: CacheConfig{
 			Expiration:      60,
