--- a/Makefile.am
+++ b/Makefile.am
@@ -6,7 +6,7 @@
 # something. DIST_SUBDIRS stores the directories where nothing is built but
 # which have makefiles with distribution information.
 #  - src (libvlccore) is nedeed by modules
-SUBDIRS = compat doc po share src modules lib bin test
+SUBDIRS = compat po share src modules lib bin test
 DIST_SUBDIRS = m4 $(SUBDIRS)
 
 EXTRA_DIST = \
@@ -117,7 +117,7 @@ core: libvlc vlc$(EXEEXT)
 doc:
 	cd doc && $(MAKE) $(AM_MAKEFLAGS) doc
 
-.PHONY: libvlc core doc
+.PHONY: libvlc core
 
 ###############################################################################
 # Building aliases
