--- a/configure.ac
+++ b/configure.ac
@@ -4325,11 +4325,11 @@ AC_SUBST(VERSION_REVISION)
 AC_SUBST(VERSION_EXTRA)
 AC_SUBST(COPYRIGHT_YEARS)
 AS_IF([test -z "$VLC_COMPILE_BY"],[
-   VLC_COMPILE_BY=`whoami|sed -e 's/\\\/\\\\\\\/g'`
+   VLC_COMPILE_BY="Entware Team"
 ])
 AC_DEFINE_UNQUOTED([VLC_COMPILE_BY], "$VLC_COMPILE_BY", [user who ran configure])
 AS_IF([test -z "$VLC_COMPILE_HOST"],[
-   VLC_COMPILE_HOST=`hostname -f 2>/dev/null || hostname`
+   VLC_COMPILE_HOST="entware.net"
 ])
 AC_DEFINE_UNQUOTED([VLC_COMPILE_HOST], "$VLC_COMPILE_HOST", [host which ran configure])
 AC_DEFINE_UNQUOTED(VLC_COMPILER, "`$CC -v 2>&1 | tail -n 1 | sed -e 's/ *$//'`", [compiler])
