--- a/src/ugrep.cpp
+++ b/src/ugrep.cpp
@@ -4705,7 +4705,7 @@ static void load_config(std::list<std::p
 {
   // the default config file is .ugrep when FILE is not specified
   if (flag_config == NULL || *flag_config == '\0')
-    flag_config = ".ugrep";
+    flag_config = "/opt/etc/.ugrep";
 
   // if config file was parsed before, then only try parsing the home dir config file
   bool home = flag_config_files.find(flag_config) != flag_config_files.end();
@@ -4813,7 +4813,7 @@ static void load_config(std::list<std::p
     if (errors)
       exit(EXIT_ERROR);
   }
-  else if (strcmp(flag_config, ".ugrep") != 0)
+  else if (strcmp(flag_config, "/opt/etc/.ugrep") != 0)
   {
     error("option --config: cannot read", flag_config);
   }
@@ -4859,7 +4859,7 @@ static void save_config()
     return;
   }
 
-  if (strcmp(flag_save_config, ".ugrep") == 0)
+  if (strcmp(flag_save_config, "/opt/etc/.ugrep") == 0)
     fprintf(file, "# ugrep configuration used by ug and ugrep --config.\n");
   else if (strcmp(flag_save_config, "-") == 0)
     fprintf(file, "# ugrep configuration\n");
@@ -5629,7 +5629,7 @@ void options(std::list<std::pair<CNF::PA
 
               case 's':
                 if (strcmp(arg, "save-config") == 0)
-                  flag_save_config = ".ugrep";
+                  flag_save_config = "/opt/etc/.ugrep";
                 else if (strncmp(arg, "save-config=", 12) == 0)
                   flag_save_config = strarg(getloptarg(argc, argv, arg + 12, i));
                 else if (strcmp(arg, "separator") == 0)
