--- a/src/github.com/filebrowser/filebrowser/cmd/root.go
+++ b/src/github.com/filebrowser/filebrowser/cmd/root.go
@@ -92,7 +92,7 @@ If you don't set "config", it will look
 
 - ./
 - $HOME/
-- /etc/filebrowser/
+- /opt/etc/filebrowser/
 
 The precedence of the configuration values are as follows:
 
@@ -389,7 +389,7 @@ func initConfig() {
 		checkErr(err)
 		v.AddConfigPath(".")
 		v.AddConfigPath(home)
-		v.AddConfigPath("/etc/filebrowser/")
+		v.AddConfigPath("/opt/etc/filebrowser/")
 		v.SetConfigName(".filebrowser")
 	} else {
 		v.SetConfigFile(cfgFile)
