--- a/dhcpd/network_utils.go
+++ b/dhcpd/network_utils.go
@@ -21,7 +21,7 @@ import (
 // Supports: Raspbian.
 func HasStaticIP(ifaceName string) (bool, error) {
 	if runtime.GOOS == "linux" {
-		body, err := ioutil.ReadFile("/etc/dhcpcd.conf")
+		body, err := ioutil.ReadFile("/opt/etc/dhcpcd.conf")
 		if err != nil {
 			return false, err
 		}
@@ -124,13 +124,13 @@ func setStaticIPDhcpdConf(ifaceName stri
 	gatewayIP := getGatewayIP(ifaceName)
 	add := updateStaticIPDhcpcdConf(ifaceName, ip, gatewayIP, ip4.String())
 
-	body, err := ioutil.ReadFile("/etc/dhcpcd.conf")
+	body, err := ioutil.ReadFile("/opt/etc/dhcpcd.conf")
 	if err != nil {
 		return err
 	}
 
 	body = append(body, []byte(add)...)
-	err = file.SafeWrite("/etc/dhcpcd.conf", body)
+	err = file.SafeWrite("/opt/etc/dhcpcd.conf", body)
 	if err != nil {
 		return err
 	}
