--- a/auto/echo/build
+++ b/auto/echo/build
@@ -7,8 +7,8 @@ $echo 'building an "echo" program'
 
 rm -f $NXT_BUILD_DIR/echo
 
-nxt_echo_test="$CC -o $NXT_BUILD_DIR/echo -O $NXT_CC_OPT
-               auto/echo/echo.c $NXT_LD_OPT"
+nxt_echo_test="cc -o $NXT_BUILD_DIR/echo -O 
+               auto/echo/echo.c"
 
 # "|| true" is to bypass "set -e" setting.
 nxt_echo_err=`$nxt_echo_test 2>&1 || true`
