From a559dcc038fbf4a5ff4b371934e75a8e1d943f1d Mon Sep 17 00:00:00 2001
From: GammaC0de <gammac0de@users.noreply.github.com>
Date: Mon, 3 May 2021 02:44:07 +0300
Subject: [PATCH 004/150] fix issue with name parsing

---
 src/pyload/core/utils/parse.py | 1 +
 1 file changed, 1 insertion(+)

--- a/src/pyload/core/utils/parse.py
+++ b/src/pyload/core/utils/parse.py
@@ -42,6 +42,7 @@ def entries(text, allow_whitespaces=Fals
 def name(text, safe_name=True):
     try:
         name = web_parse.name(text, safe_name=safe_name)
+        name = os.path.basename(name)
     except Exception as exc:
         name = os.path.basename(text).strip()
         if safe_name:
