From df8619a026c269d60efbedf494e39f80b8a764ab Mon Sep 17 00:00:00 2001
From: GammaC0de <gammac0de@users.noreply.github.com>
Date: Tue, 1 Jun 2021 01:57:52 +0300
Subject: [PATCH 073/150] do not display warning if english

---
 src/pyload/core/__init__.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/src/pyload/core/__init__.py
+++ b/src/pyload/core/__init__.py
@@ -263,7 +263,8 @@ class Core:
         try:
             self.set_language(lang)
         except IOError as exc:
-            self.log.warning(exc, exc_info=self.debug > 1, stack_info=self.debug > 2)
+            if lang != "en":
+                self.log.warning(exc, exc_info=self.debug > 1, stack_info=self.debug > 2)
             self._set_language(self.LOCALE_DOMAIN, fallback=True)
 
     # def _setup_niceness(self):
