From f69ad2deb245f142fd710b837913a15350478932 Mon Sep 17 00:00:00 2001
From: GammaC0de <gammac0de@users.noreply.github.com>
Date: Mon, 14 Jun 2021 00:07:52 +0300
Subject: [PATCH 081/150] chdir to data directory

---
 src/pyload/core/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/pyload/core/__init__.py
+++ b/src/pyload/core/__init__.py
@@ -93,8 +93,8 @@ class Core:
 
         self._init_config(userdir, tempdir, storagedir, debug)
         self._init_log()
-
         self._init_database(reset and not dry)
+        os.chdir(os.path.join(self.userdir, "data"))
         self._init_network()
         self._init_api()
         self._init_managers()
