From 2226f82cc4a5bb6b7dff4349ffadb5d8017f5c77 Mon Sep 17 00:00:00 2001
From: GammaC0de <gammac0de@users.noreply.github.com>
Date: Wed, 16 Jun 2021 00:27:47 +0300
Subject: [PATCH 093/150] [SolidfilesCom] sync with stable

---
 .../plugins/downloaders/SolidfilesCom.py      | 20 ++++++++-----------
 1 file changed, 8 insertions(+), 12 deletions(-)

--- a/src/pyload/plugins/downloaders/SolidfilesCom.py
+++ b/src/pyload/plugins/downloaders/SolidfilesCom.py
@@ -7,11 +7,13 @@ from ..base.simple_downloader import Sim
 # Test links:
 #   http://www.solidfiles.com/d/609cdb4b1b
 
+import re
+
 
 class SolidfilesCom(SimpleDownloader):
     __name__ = "SolidfilesCom"
     __type__ = "downloader"
-    __version__ = "0.08"
+    __version__ = "0.09"
     __status__ = "testing"
 
     __pattern__ = r"http://(?:www\.)?solidfiles\.com\/d/\w+"
@@ -30,21 +32,15 @@ class SolidfilesCom(SimpleDownloader):
         ("GammaC0de", "nitzo2001[AT]yahoo[DOT]com"),
     ]
 
-    NAME_PATTERN = r'<i class="icon filetype-icon.+?</i>\s*<h1>\s*(?P<N>.+?)\s*</h1>'
-    SIZE_PATTERN = r'copy-text="viewer\.node\.viewUrl"></copy-button>\s*(?P<S>[\d.,]+) (?P<U>[\w_^]+)'
+    NAME_PATTERN = r'<h1 class="node-name">(?P<N>.+?)</h1>'
+    SIZE_PATTERN = r"</copy-button>\s*(?P<S>[\d.,]+) (?P<U>[\w_^]+)"
     OFFLINE_PATTERN = r"<h1>404"
 
-    LINK_FREE_PATTERN = r'<a href="(https?://.*\.solidfilesusercontent\.com/.+?)"'
-
     def setup(self):
         self.multi_dl = True
         self.chunk_limit = 1
 
     def handle_free(self, pyfile):
-        action, inputs = self.parse_html_form('action="/v/')
-        if not action or not inputs:
-            self.error(self._("Free download form not found"))
-
-        self.data = self.load(self.fixurl(action), post=inputs)
-
-        super().handle_free(pyfile)
+        m = re.search(r'"downloadUrl":"(.+?)"', self.data)
+        if m is not None:
+            self.link = m.group(1)
