From f3e71498652362a6413f931dda764287ecdca789 Mon Sep 17 00:00:00 2001
From: GammaC0de <gammac0de@users.noreply.github.com>
Date: Tue, 17 Aug 2021 11:28:04 +0300
Subject: [PATCH 135/150] remove deprecated `user` variable

---
 src/pyload/plugins/base/hoster.py | 3 ---
 1 file changed, 3 deletions(-)

--- a/src/pyload/plugins/base/hoster.py
+++ b/src/pyload/plugins/base/hoster.py
@@ -154,7 +154,6 @@ class BaseHoster(BasePlugin):
             self.load_account()  # TODO: Move to PluginThread in 0.6.x
         else:
             self.account = False
-            self.user = None  # TODO: Remove in 0.6.x
 
         try:
             self.req.close()
@@ -184,11 +183,9 @@ class BaseHoster(BasePlugin):
 
         if not self.account:
             self.account = False
-            self.user = None  # TODO: Remove in 0.6.x
 
         else:
             self.account.choose()
-            self.user = self.account.user  # TODO: Remove in 0.6.x
             if self.account.user is None:
                 self.account = False
 
