--- a/flask_compress/flask_compress.py
+++ b/flask_compress/flask_compress.py
@@ -11,7 +11,7 @@ from io import BytesIO
 
 from collections import defaultdict
 
-import brotli
+#import brotli
 from flask import request, after_this_request, current_app
 
 
@@ -229,9 +229,9 @@ class Compress(object):
         elif algorithm == 'deflate':
             return zlib.compress(response.get_data(),
                                  app.config['COMPRESS_DEFLATE_LEVEL'])
-        elif algorithm == 'br':
-            return brotli.compress(response.get_data(),
-                                   mode=app.config['COMPRESS_BR_MODE'],
-                                   quality=app.config['COMPRESS_BR_LEVEL'],
-                                   lgwin=app.config['COMPRESS_BR_WINDOW'],
-                                   lgblock=app.config['COMPRESS_BR_BLOCK'])
+#        elif algorithm == 'br':
+#            return brotli.compress(response.get_data(),
+#                                   mode=app.config['COMPRESS_BR_MODE'],
+#                                   quality=app.config['COMPRESS_BR_LEVEL'],
+#                                   lgwin=app.config['COMPRESS_BR_WINDOW'],
+#                                   lgblock=app.config['COMPRESS_BR_BLOCK'])
--- a/setup.py
+++ b/setup.py
@@ -18,8 +18,7 @@ setup(
     include_package_data=True,
     platforms='any',
     install_requires=[
-        'flask',
-        'brotli'
+        'flask'
     ],
     classifiers=[
         'Environment :: Web Environment',
