--- a/spa/plugins/alsa/acp/compat.c
+++ b/spa/plugins/alsa/acp/compat.c
@@ -277,7 +277,7 @@ char *get_data_path(const char *data_dir
     }
 
     SPA_FOR_EACH_ELEMENT_VAR(subpaths, subpath) {
-	spa_autofree char *path = spa_aprintf("/etc/%s/%s", *subpath, data_type);
+	spa_autofree char *path = spa_aprintf("/opt/etc/%s/%s", *subpath, data_type);
 	if ((result = try_path(fname, path)) != NULL)
 	    return result;
     }
--- a/src/modules/module-filter-chain/ladspa_plugin.c
+++ b/src/modules/module-filter-chain/ladspa_plugin.c
@@ -221,7 +221,7 @@ struct fc_plugin *load_ladspa_plugin(con
 
 		search_dirs = getenv("LADSPA_PATH");
 		if (!search_dirs)
-			search_dirs = "/usr/lib64/ladspa:/usr/lib/ladspa:" LIBDIR;
+			search_dirs = "/opt/lib/ladspa:/opt/usr/lib/ladspa:" LIBDIR;
 
 		/*
 		 * set the errno for the case when `ladspa_handle_load_by_path()`
--- a/src/modules/module-protocol-native/local-socket.c
+++ b/src/modules/module-protocol-native/local-socket.c
@@ -23,7 +23,7 @@
 #include <pipewire/pipewire.h>
 #include <spa/utils/json.h>
 
-#define DEFAULT_SYSTEM_RUNTIME_DIR "/run/pipewire"
+#define DEFAULT_SYSTEM_RUNTIME_DIR "/opt/var/run/pipewire"
 
 PW_LOG_TOPIC_EXTERN(mod_topic);
 #define PW_LOG_TOPIC_DEFAULT mod_topic
--- a/src/modules/module-pipe-tunnel.c
+++ b/src/modules/module-pipe-tunnel.c
@@ -116,8 +116,8 @@
 
 #define NAME "pipe-tunnel"
 
-#define DEFAULT_CAPTURE_FILENAME	"/tmp/fifo_input"
-#define DEFAULT_PLAYBACK_FILENAME	"/tmp/fifo_output"
+#define DEFAULT_CAPTURE_FILENAME	"/opt/tmp/fifo_input"
+#define DEFAULT_PLAYBACK_FILENAME	"/opt/tmp/fifo_output"
 
 #define DEFAULT_FORMAT "S16"
 #define DEFAULT_RATE 48000
