--- a/mpcgain/CMakeLists.txt
+++ b/mpcgain/CMakeLists.txt
@@ -1,13 +1,9 @@
 FIND_PATH(REPLAY_GAIN_INCLUDE_DIR replaygain/gain_analysis.h
-	/usr/include
-	/usr/local/include
-	${libmpc_SOURCE_DIR}/../libreplaygain/include
+	${libmpc_SOURCE_DIR}/../libreplaygain/include
 )
 
 FIND_LIBRARY(REPLAY_GAIN_LIBRARY NAMES replaygain replaygain_static PATHS
-	/usr/lib
-	/usr/local/lib
-	${libmpc_SOURCE_DIR}/../libreplaygain/bin/src/release
+	${libmpc_SOURCE_DIR}/../libreplaygain/bin/src/release
 )
 
 include_directories(${libmpc_SOURCE_DIR}/include)
