--- a/librz/core/golang.c
+++ b/librz/core/golang.c
@@ -1129,7 +1129,8 @@ static bool decode_lui_set_addr(RzCore *
 	return true;
 }
 
-go_asm_pattern_define(mips, 32, any, "\x00\x00\x00\x00", "\x00\x00\x00\x00", false);
+go_asm_pattern_define(mipsbe, 32, any, "\x00\x00\x00\x00", "\x00\x00\x00\x00", false);
+go_asm_pattern_define(mipsle, 32, any, "\x00\x00\x00\x00", "\x00\x00\x00\x00", false);
 go_asm_pattern_define(mipsbe, 32, addiu, "\x24\x00\x00\x00", "\xff\x00\x00\x00", false);
 go_asm_pattern_define(mipsle, 32, addiu, "\x00\x00\x00\x24", "\x00\x00\x00\xff", false);
 go_asm_pattern_define(mipsbe, 32, lui, "\x3c\x00\x00\x00", "\xff\x00\x00\x00", true);
@@ -1142,7 +1143,7 @@ static GoSignature go_mipsle32_lui_addiu
 	// addiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsle, 32, addiu), &decode_val_add_addr },
 	// sw    v0, 0x08(at)
-	{ &go_asm_pattern_name(mips, 32, any), NULL },
+	{ &go_asm_pattern_name(mipsle, 32, any), NULL },
 	// addiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsle, 32, addiu), &decode_val_set_size },
 };
@@ -1151,7 +1152,7 @@ static GoSignature go_mipsle32_addiu_sw_
 	// addiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsle, 32, addiu), &decode_val_set_size },
 	// sw    v0, 0x08(at)
-	{ &go_asm_pattern_name(mips, 32, any), NULL },
+	{ &go_asm_pattern_name(mipsle, 32, any), NULL },
 	// lui   v0, high_string_offset
 	{ &go_asm_pattern_name(mipsle, 32, lui), &decode_lui_set_addr },
 	// addiu v0, v0, low_string_offset
@@ -1173,7 +1174,7 @@ static GoSignature go_mipsle32_table_sig
 	// addiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsle, 32, addiu), &decode_val_add_addr },
 	// sw    v0, 0x08(at)
-	{ &go_asm_pattern_name(mips, 32, any), &decode_from_table },
+	{ &go_asm_pattern_name(mipsle, 32, any), &decode_from_table },
 };
 
 // ---- BE ----
@@ -1183,7 +1184,7 @@ static GoSignature go_mipsbe32_lui_addiu
 	// addiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsbe, 32, addiu), &decode_val_add_addr },
 	// sw    v0, 0x08(at)
-	{ &go_asm_pattern_name(mips, 32, any), NULL },
+	{ &go_asm_pattern_name(mipsbe, 32, any), NULL },
 	// addiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsbe, 32, addiu), &decode_val_set_size },
 };
@@ -1192,7 +1193,7 @@ static GoSignature go_mipsbe32_addiu_sw_
 	// addiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsbe, 32, addiu), &decode_val_set_size },
 	// sw    v0, 0x08(at)
-	{ &go_asm_pattern_name(mips, 32, any), NULL },
+	{ &go_asm_pattern_name(mipsbe, 32, any), NULL },
 	// lui   v0, high_string_offset
 	{ &go_asm_pattern_name(mipsbe, 32, lui), &decode_lui_set_addr },
 	// addiu v0, v0, low_string_offset
@@ -1214,7 +1215,7 @@ static GoSignature go_mipsbe32_table_sig
 	// addiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsbe, 32, addiu), &decode_val_add_addr },
 	// sw    v0, 0x08(at)
-	{ &go_asm_pattern_name(mips, 32, any), &decode_from_table },
+	{ &go_asm_pattern_name(mipsbe, 32, any), &decode_from_table },
 };
 
 static ut32 golang_recover_string_mips32(GoStrRecover *ctx) {
@@ -1244,7 +1245,8 @@ static ut32 golang_recover_string_mips32
 	return 4;
 }
 
-go_asm_pattern_define(mips, 64, any, "\x00\x00\x00\x00", "\x00\x00\x00\x00", false);
+go_asm_pattern_define(mipsbe, 64, any, "\x00\x00\x00\x00", "\x00\x00\x00\x00", false);
+go_asm_pattern_define(mipsle, 64, any, "\x00\x00\x00\x00", "\x00\x00\x00\x00", false);
 go_asm_pattern_define(mipsbe, 64, daddu, "\x00\x00\x00\x2d", "\x00\x00\x00\xff", false);
 go_asm_pattern_define(mipsle, 64, daddu, "\x2d\x00\x00\x00", "\xff\x00\x00\x00", false);
 go_asm_pattern_define(mipsbe, 64, daddiu, "\x64\x00\x00\x00", "\xfc\x00\x00\x00", false);
@@ -1263,7 +1265,7 @@ static GoSignature go_mipsle64_lui_daddu
 	// daddiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsle, 64, daddiu), &decode_val_add_addr },
 	// sd    v0, 8(at)
-	{ &go_asm_pattern_name(mips, 64, any), NULL },
+	{ &go_asm_pattern_name(mipsle, 64, any), NULL },
 	// daddiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsle, 64, move), &decode_val_set_size },
 };
@@ -1272,7 +1274,7 @@ static GoSignature go_mipsle64_daddiu_sd
 	// daddiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsle, 64, move), &decode_val_set_size },
 	// sd    v0, 8(at)
-	{ &go_asm_pattern_name(mips, 64, any), NULL },
+	{ &go_asm_pattern_name(mipsle, 64, any), NULL },
 	// lui   v0, high_string_offset
 	{ &go_asm_pattern_name(mipsle, 64, lui), &decode_lui_set_addr },
 	// daddu v0, v0, gp
@@ -1300,7 +1302,7 @@ static GoSignature go_mipsle64_table_sig
 	// daddiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsle, 64, daddiu), &decode_val_add_addr },
 	// sd    v0, 8(at)
-	{ &go_asm_pattern_name(mips, 64, any), &decode_from_table },
+	{ &go_asm_pattern_name(mipsle, 64, any), &decode_from_table },
 };
 
 // ---- BE ----
@@ -1312,7 +1314,7 @@ static GoSignature go_mipsbe64_lui_daddu
 	// daddiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsbe, 64, daddiu), &decode_val_add_addr },
 	// sd    v0, 8(at)
-	{ &go_asm_pattern_name(mips, 64, any), NULL },
+	{ &go_asm_pattern_name(mipsbe, 64, any), NULL },
 	// daddiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsbe, 64, move), &decode_val_set_size },
 };
@@ -1321,7 +1323,7 @@ static GoSignature go_mipsbe64_daddiu_sd
 	// daddiu v0, zero, string_size
 	{ &go_asm_pattern_name(mipsbe, 64, move), &decode_val_set_size },
 	// sd    v0, 8(at)
-	{ &go_asm_pattern_name(mips, 64, any), NULL },
+	{ &go_asm_pattern_name(mipsbe, 64, any), NULL },
 	// lui   v0, high_string_offset
 	{ &go_asm_pattern_name(mipsbe, 64, lui), &decode_lui_set_addr },
 	// daddu v0, v0, gp
@@ -1349,7 +1351,7 @@ static GoSignature go_mipsbe64_table_sig
 	// daddiu v0, v0, low_string_offset
 	{ &go_asm_pattern_name(mipsbe, 64, daddiu), &decode_val_add_addr },
 	// sd    v0, 8(at)
-	{ &go_asm_pattern_name(mips, 64, any), &decode_from_table },
+	{ &go_asm_pattern_name(mipsbe, 64, any), &decode_from_table },
 };
 
 static ut32 golang_recover_string_mips64(GoStrRecover *ctx) {
