--- a/examples/etc/turnserver.conf
+++ b/examples/etc/turnserver.conf
@@ -60,6 +60,7 @@
 #listening-ip=172.17.19.101
 #listening-ip=10.207.21.238
 #listening-ip=2607:f0d0:1002:51::4
+listening-ip=127.0.0.1
 
 # Auxiliary STUN/TURN server listening endpoint.
 # Aux servers have almost full TURN and STUN functionality.
@@ -77,6 +78,7 @@
 #
 #aux-server=172.17.19.110:33478
 #aux-server=[2607:f0d0:1002:51::4]:33478
+aux-server=127.0.0.1
 
 # (recommended for older Linuxes only)
 # Automatically balance UDP traffic over auxiliary servers (if configured).
@@ -104,6 +106,7 @@
 #
 #relay-ip=172.17.19.105
 #relay-ip=2607:f0d0:1002:51::5
+relay-ip=127.0.0.1
 
 # For Amazon EC2 users:
 #
@@ -282,10 +285,9 @@
 
 # SQLite database file name.
 #
-# The default file name is /var/db/turndb or /usr/local/var/db/turndb or
-# /var/lib/turn/turndb.
+# The default file name is /opt/var/lib/turn/turndb.
 #
-#userdb=/var/db/turndb
+userdb=/opt/var/lib/turn/turndb
 
 # PostgreSQL database connection string in the case that you are using PostgreSQL
 # as the user database.
@@ -456,14 +458,14 @@
 # configuration file.
 # Use PEM file format.
 #
-#cert=/usr/local/etc/turn_server_cert.pem
+cert=/opt/etc/turnserver/turn_server_cert.pem
 
 # Private key file.
 # Use an absolute path or path relative to the
 # configuration file.
 # Use PEM file format.
 #
-#pkey=/usr/local/etc/turn_server_pkey.pem
+pkey=/opt/etc/turnserver/turn_server_pkey.pem
 
 # Private key file password, if it is in encoded format.
 # This option has no default value.
@@ -481,7 +483,7 @@
 # certificate is not checked.
 #
 # Example:
-#CA-file=/etc/ssh/id_rsa.cert
+#CA-file=/opt/etc/ssh/id_rsa.cert
 
 # Curve name for EC ciphers, if supported by OpenSSL
 # library (TLS and DTLS). The default value is prime256v1,
@@ -509,11 +511,10 @@
 # the configured log file. With this option everything will
 # go to the configured log only (unless the log file itself is stdout).
 #
-#no-stdout-log
+no-stdout-log
 
 # Option to set the log file name.
-# By default, the turnserver tries to open a log file in
-# /var/log, /var/tmp, /tmp and the current directory
+# By default, the turnserver tries to open a log file in /opt/var/log
 # (Whichever file open operation succeeds first will be used).
 # With this option you can set the definite log file name.
 # The special names are "stdout" and "-" - they will force everything
@@ -522,7 +523,7 @@
 # In the runtime, the logfile can be reset with the SIGHUP signal
 # to the turnserver process.
 #
-#log-file=/var/tmp/turn.log
+log-file=/opt/var/log/turn.log
 
 # Option to redirect all log output into system log (syslog).
 #
@@ -644,10 +645,9 @@
 # allowed-peer-ip=83.166.68.45
 
 # File name to store the pid of the process.
-# Default is /var/run/turnserver.pid (if superuser account is used) or
-# /var/tmp/turnserver.pid .
+# Default is /opt/var/run/turnserver.pid (if superuser account is used).
 #
-#pidfile="/var/run/turnserver.pid"
+pidfile="/opt/var/run/turnserver.pid"
 
 # Require authentication of the STUN Binding request.
 # By default, the clients are allowed anonymous access to the STUN Binding functionality.
@@ -670,12 +670,12 @@
 # User name to run the process. After the initialization, the turnserver process
 # will attempt to change the current user ID to that user.
 #
-#proc-user=<user-name>
+proc-user=root
 
 # Group name to run the process. After the initialization, the turnserver process
 # will attempt to change the current group ID to that group.
 #
-#proc-group=<group-name>
+proc-group=root
 
 # Turn OFF the CLI support.
 # By default it is always ON.
@@ -686,7 +686,7 @@
 #Local system IP address to be used for CLI server endpoint. Default value
 # is 127.0.0.1.
 #
-#cli-ip=127.0.0.1
+cli-ip=127.0.0.1
 
 # CLI server port. Default is 5766.
 #
@@ -702,26 +702,26 @@
 #
 # Or unsecure form for the same password:
 #
-#cli-password=qwerty
+cli-password=qwerty
 
 # Enable Web-admin support on https. By default it is Disabled.
 # If it is enabled it also enables a http a simple static banner page
 # with a small reminder that the admin page is available only on https.
 #
-#web-admin
+web-admin
 
 # Local system IP address to be used for Web-admin server endpoint. Default value is 127.0.0.1.
 #
-#web-admin-ip=127.0.0.1
+web-admin-ip=127.0.0.1
 
 # Web-admin server port. Default is 8080.
 #
-#web-admin-port=8080
+web-admin-port=8080
 
 # Web-admin server listen on STUN/TURN worker threads
 # By default it is disabled for security resons! (Not recommended in any production environment!)
 #
-#web-admin-listen-on-workers
+web-admin-listen-on-workers
 
 #acme-redirect=http://redirectserver/.well-known/acme-challenge/
 # Redirect ACME, i.e. HTTP GET requests matching '^/.well-known/acme-challenge/(.*)' to '<URL>$1'.
