--- a/cmd/root.go
+++ b/cmd/root.go
@@ -167,7 +167,7 @@ func init() {
 		conf.InitConfig(cfgFile)
 	})
 
-	rootCmd.PersistentFlags().StringVarP(&cfgFile, "configfile", "c", "", `config file (default "./navidrome.toml")`)
+	rootCmd.PersistentFlags().StringVarP(&cfgFile, "configfile", "c", "", `config file (default "/opt/etc/navidrome.toml")`)
 	rootCmd.PersistentFlags().BoolVarP(&noBanner, "nobanner", "n", false, `don't show banner`)
 	rootCmd.PersistentFlags().String("musicfolder", viper.GetString("musicfolder"), "folder where your music is stored")
 	rootCmd.PersistentFlags().String("datafolder", viper.GetString("datafolder"), "folder to store application data (DB), needs write access")
--- a/conf/configuration.go
+++ b/conf/configuration.go
@@ -270,9 +270,9 @@ func AddHook(hook func()) {
 }
 
 func init() {
-	viper.SetDefault("musicfolder", filepath.Join(".", "music"))
+	viper.SetDefault("musicfolder", filepath.Join("/opt/data", "music"))
 	viper.SetDefault("cachefolder", "")
-	viper.SetDefault("datafolder", ".")
+	viper.SetDefault("datafolder", "/opt/var/lib/navidrome")
 	viper.SetDefault("loglevel", "info")
 	viper.SetDefault("address", "0.0.0.0")
 	viper.SetDefault("port", 4533)
@@ -372,7 +372,7 @@ func InitConfig(cfgFile string) {
 		viper.SetConfigFile(cfgFile)
 	} else {
 		// Search config in local directory with name "navidrome" (without extension).
-		viper.AddConfigPath(".")
+		viper.AddConfigPath("/opt/etc")
 		viper.SetConfigName("navidrome")
 	}
 
