From 63a74bef9e9f639bc199255db9d452d96f01d11a Mon Sep 17 00:00:00 2001
From: Max Lv <mlv@nvidia.com>
Date: Tue, 21 Jul 2020 08:26:57 +0800
Subject: [PATCH 3/5] Fix the VPU option

---
 log_android.go | 17 +++++++++++------
 main.go        |  2 +-
 2 files changed, 12 insertions(+), 7 deletions(-)

diff --git a/log_android.go b/log_android.go
index 9030533..aa2f3b8 100644
--- a/src/github.com/shadowsocks/v2ray-plugin/log_android.go
+++ b/src/github.com/shadowsocks/v2ray-plugin/log_android.go
@@ -26,17 +26,22 @@ var (
 	ctag = C.CString("v2ray")
 )
 
-type androidLogger struct {}
+type androidLogger struct{}
+
 func (l *androidLogger) Handle(msg log.Message) {
-	var priority = C.ANDROID_LOG_FATAL	// this value should never be used in client mode
+	var priority = C.ANDROID_LOG_FATAL // this value should never be used in client mode
 	var message string
 	switch msg := msg.(type) {
 	case *log.GeneralMessage:
 		switch msg.Severity {
-		case log.Severity_Error:   priority = C.ANDROID_LOG_ERROR
-		case log.Severity_Warning: priority = C.ANDROID_LOG_WARN
-		case log.Severity_Info:    priority = C.ANDROID_LOG_INFO
-		case log.Severity_Debug:   priority = C.ANDROID_LOG_DEBUG
+		case log.Severity_Error:
+			priority = C.ANDROID_LOG_ERROR
+		case log.Severity_Warning:
+			priority = C.ANDROID_LOG_WARN
+		case log.Severity_Info:
+			priority = C.ANDROID_LOG_INFO
+		case log.Severity_Debug:
+			priority = C.ANDROID_LOG_DEBUG
 		}
 		message = serial.ToString(msg.Content)
 	default:
diff --git a/main.go b/main.go
index a6979e1..cf71a4a 100644
--- a/src/github.com/shadowsocks/v2ray-plugin/main.go
+++ b/src/github.com/shadowsocks/v2ray-plugin/main.go
@@ -329,7 +329,7 @@ func startV2Ray() (core.Server, error) {
 			*fastOpen = true
 		}
 
-		if _, b := opts.Get("vpn"); b {
+		if _, b := opts.Get("V"); b {
 			*vpn = true
 		}
 
-- 
2.20.1

