--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
-CC=gcc
-CFLAGS=-Wall -I/usr/local/include -O3
+CC ?= $(CC)
+CFLAGS ?= -Wall
 
+LDFLAGS ?=
 LIBS=-pthread `pkg-config libevent --libs-only-L` -levent -levent_pthreads
 
 ifeq ($(shell uname -m),armv7l)
@@ -36,6 +37,6 @@ cmdline.h: options.ggo
 cmdline.c: cmdline.h
 
 rtlmux: $(OBJS)
-	$(CC) $(CFLAGS) -o $@ $^ $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $^ $(LDFLAGS) $(LIBS)
 
 # DO NOT DELETE
