--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS= pango pango-view examples docs tools tests win32
+SUBDIRS= pango pango-view examples docs tools win32
 
 meson_dist_files = \
 	pango/pango-features.h.meson \
--- a/Makefile.in
+++ b/Makefile.in
@@ -412,7 +412,7 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = pango pango-view examples docs tools tests win32
+SUBDIRS = pango pango-view examples docs tools win32
 meson_dist_files = \
 	pango/pango-features.h.meson \
 	pango-view/meson.build \
