--- a/server/CMakeLists.txt
+++ b/server/CMakeLists.txt
@@ -129,14 +129,6 @@ else()
   install(
     FILES etc/plug-ins/meta_mpd.py etc/plug-ins/meta_mopidy.py
           etc/plug-ins/meta_librespot-java.py
-    PERMISSIONS
-      OWNER_READ
-      OWNER_WRITE
-      OWNER_EXECUTE
-      GROUP_READ
-      GROUP_EXECUTE
-      WORLD_READ
-      WORLD_EXECUTE
     DESTINATION ${CMAKE_INSTALL_DATADIR}/snapserver/plug-ins/)
   # install(FILES ../debian/snapserver.service DESTINATION
   # ${SYSTEMD_SERVICES_INSTALL_DIR})
